tests/erlang_tests/float_bif.erl

```diff
@@ -0,0 +1,45 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2025 Jakub Gonet <jakub.gonet@swmansion.com>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(float_bif).
+
+-export([start/0]).
+% For remote function calls
+-export([id/1, check_one/1]).
+
+-define(ID(Arg), ?MODULE:id(Arg)).
+
+start() ->
+    true = 1.0 =:= float(?ID(1)),
+    true = 1.0 =:= float(?ID(1.0)),
+    {'EXIT', {badarg, _stack}} = catch float(?ID("0")),
```

Comment:
OTP 21 to 23 require parentheses I believe.

```suggestion
    {'EXIT', {badarg, _stack}} = (catch float(?ID("0"))),
```

Reply:
...and a variable should be uppercase.